src/common/meta/src/ddl/create_view.rs

```diff
@@ -0,0 +1,276 @@
+                    view_name: self.creator.data.table_ref().to_string(),
+                }
+            );
+
+            let exists_view_id = value.table_id();
+
+            if !expr.or_replace {
+                return Ok(Status::done_with_output(exists_view_id));
+            }
+            view_id = Some(exists_view_id);
+        }
+
+        if let Some(view_id) = view_id {
+            let view_info_value = self
+                .context
+                .table_metadata_manager
+                .table_info_manager()
+                .get(view_id)
+                .await?
+                .with_context(|| error::TableInfoNotFoundSnafu {
+                    table: self.creator.data.table_ref().to_string(),
+                })?;
+
+            // Ensure the exists one is view, we can't replace a table.
+            ensure!(
+                view_info_value.table_info.table_type == TableType::View,
+                error::TableAlreadyExistsSnafu {
+                    table_name: self.creator.data.table_ref().to_string(),
+                }
+            );
+
+            self.creator.set_allocated_metadata(view_id, true);
+        } else {
+            // Allocate the new `view_id`.
+            let TableMetadata { table_id, .. } = self
+                .context
+                .table_metadata_allocator
+                .create_view(
+                    &TableMetadataAllocatorContext {
+                        cluster_id: self.creator.data.cluster_id,
+                    },
+                    &None,
+                )
+                .await?;
+            self.creator.set_allocated_metadata(table_id, false);
+        }
+
+        self.creator.data.state = CreateViewState::CreateMetadata;
+
+        Ok(Status::executing(true))
+    }
+
+    /// Creates view metadata
+    ///
+    /// Abort(not-retry):
+    /// - Failed to create view metadata.
+    async fn on_create_metadata(&mut self, ctx: &ProcedureContext) -> Result<Status> {
+        let view_id = self.view_id();
+        let manager = &self.context.table_metadata_manager;
+
+        if self.need_update() {
+            // Retrieve the current view info and try to update it.
+            let current_view_info = manager
+                .view_info_manager()
+                .get(view_id)
+                .await?
+                .with_context(|| error::ViewNotFoundSnafu {
+                    view_name: self.creator.data.table_ref().to_string(),
+                })?;
+            let new_logical_plan = self.creator.data.task.raw_logical_plan().clone();
+            manager
+                .update_view_info(view_id, &current_view_info, new_logical_plan)
+                .await?;
+
+            info!("Updated view metadata for view {view_id}");
+        } else {
+            let raw_view_info = self.view_info().clone();
+            manager
+                .create_view_metadata(raw_view_info, self.creator.data.task.raw_logical_plan())
```

Comment:
Question: What if `raw_view_info` is altered in the future, will the user need to manually alter the `View`? Or the system will handle it automatically?

Reply:
@WenyXu We don't process such cases when the original query tables or views are altered after creating the view.  Confirms that MySQL doesn't process it too.